dts/arc/mem.h

```diff
@@ -0,0 +1,13 @@
+#ifndef __DT_BINDING_ST_MEM_H
+#define __DT_BINDING_ST_MEM_H
+
+#define __SIZE_K(x) (x * 1024)
+
+#if defined(CONFIG_SOC_QUARK_SE_C1000_SS)
+#define DT_FLASH_SIZE		__SIZE_K(152)
```

Comment:
can we please skip the mem.h and define this directly in the board dts? I would do the same for x86, this way we will have all information in one place, I know it was done like this for some arm boards but I do not think it is justified here.

Reply:
Its partially a question if the memory & flash sizes are a function of the board or SoC.

Reply:
I think reason is quark_se_c1000 soc based borads arduino_101 and tinytile has different flash size so it make sense to move it to the board. so it is a function of board